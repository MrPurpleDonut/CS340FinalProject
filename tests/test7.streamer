funk void printFile(text path)
{
  text contents << path;
  print contents;
}

printFile(input "Enter file path: ");
