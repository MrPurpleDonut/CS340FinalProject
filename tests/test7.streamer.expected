Program parsed correctly!
---------------------------
--------Printing AST-------
---------------------------
Program
    Function List
        Function declaration: printFile
            Type: void
            Parameter List
                Parameter: path
                    Type: text
            Statement List
                Declaration statement: contents
                    Type: text
                    File Read into variable: contents
                        Identifier: path
                Print Statement
                    Identifier: contents
    Statement List
        Statement as Expression
            Function call: printFile
                Expression List
                    Input expression
                        Quoted string: "Enter file path: "
